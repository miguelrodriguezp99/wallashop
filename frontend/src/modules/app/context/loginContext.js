import { createContext, useState } from "react";

// Este es el que tenemos que consumir
export const LogginContext = createContext();

// Este es el que nos provee de acceso al contexto
export function LogginProvider({ children }) {
	const [token, setToken] = useState(null);
	const [image, setImage] = useState(null);
	const [user, setUser] = useState({});

	return (
		<LogginContext.Provider
			value={{
				token,
				setToken,
				image,
				setImage,
				user,
				setUser,
			}}
		>
			{children}
		</LogginContext.Provider>
	);
}
