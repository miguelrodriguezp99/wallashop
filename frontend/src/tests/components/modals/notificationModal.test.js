import React from 'react';
import NotificationModal from '../../../modules/app/components/modals/notificationModal';
import { MemoryRouter } from 'react-router-dom'; 
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { LogginContext } from '../../../modules/app/context/loginContext';

const fakeLogginContext = {
    user: { id: 1 }, 
    token: "testToken",
};
  
  describe("NotificationModal", () => {
    it("renders correctly", () => {
      const tree = renderer.create(
        <MemoryRouter>
          <LogginContext.Provider value={fakeLogginContext}>
            <NotificationModal />
          </LogginContext.Provider>
        </MemoryRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });