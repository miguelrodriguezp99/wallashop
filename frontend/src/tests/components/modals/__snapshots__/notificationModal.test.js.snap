// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationModal renders correctly 1`] = `
Array [
  <button
    className="group tap-highlight-transparent transition-transform-colors-opacity data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-primary/20 data-[hover=true]:opacity-hover text-black"
    disabled={false}
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
    type="button"
  >
    Notificaciones
  </button>,
  <div
    className="mt-20"
  />,
]
`;
